content/ADR-284-private-voice-chat.md

```diff
@@ -0,0 +1,362 @@
+This call ID can be used to identify the call and end it (if the user wants to). The callee will be notified through the updates subscription mechanism mentioned in the _Subscribing private voice chats updates_ section.
+
+Here's the interaction flow of the Explorer and the Social Service:
+
+```mermaid
+  sequenceDiagram
+    participant U1 as Caller
+    participant SS as Social Service
+
+    U1->>SS: Start private voice chat with callee
+    alt Caller or callee allow only calls from friends and they're not friends
+      SS-->>U1: Forbidden error
+    else Caller or callee are ringing another user
+      SS-->>U1: Conflicting error
+    else Caller or callee are in another private voice chat
+      SS-->>U1: Conflicting error
+    else Callee and caller can start a conversation
+      SS-->>U1: Call id
+    end
+```
+
+The RPC call is specified in the protocol as:
+
+```protobuf
+message StartPrivateVoiceChatPayload {
+  User callee = 1;
+}
+
+message StartPrivateVoiceChatResponse {
+  message Ok {
+    string call_id = 1;
+  }
+
+  oneof response {
+    Ok ok = 1;
+    InternalServerError internal_server_error = 2;
+    ConflictingError conflicting_error = 4;
+    ForbiddenError forbidden_error = 5;
+  }
+}
+
+rpc StartPrivateVoiceChat(StartPrivateVoiceChatPayload) returns (StartPrivateVoiceChatResponse) {}
+```
+
+### Rejecting a private voice chat
+
+Rejecting a private voice chat is done by performing the `RejectPrivateVoiceChat` RPC call to the Social Service with the call id of the ringing voice chat.
+If successful, it will return without errors and will end the ringing process. The caller will be notified through the updates subscription mechanism mentioned in the _Subscribing private voice chats updates_ section.
+
+Here's the interaction flow of the Explorer and the Social Service:
+
+```mermaid
+sequenceDiagram
+  actor U2 as Callee
+  participant SS as Social Server
+
+  U2->>SS: Reject call
+  alt The call doesn't exist
+	  SS-->>U2: Not found error
+	else The call exists
+	  SS-->>U2: Ok
+	end
+```
+
+The RPC call is specified in the protocol as:
+
+```protobuf
+message RejectPrivateVoiceChatPayload {
+  string call_id = 1;
+}
+
+message RejectPrivateVoiceChatResponse {
+  message Ok {
+    string call_id = 1;
+  }
+
+  oneof response {
+    Ok ok = 1;
+    InternalServerError internal_server_error = 2;
+    NotFoundError not_found = 4;
+  }
+}
+
+rpc RejectPrivateVoiceChat(RejectPrivateVoiceChatPayload) returns (RejectPrivateVoiceChatResponse) {}
+```
+
+### Accepting a private voice chat
+
+Accepting a private voice chat is done by performing the `AcceptPrivateVoiceChat` RPC call to the Social Service with the call id of the ringing voice chat.
+If successful, it will return return the connection url, which is crafted using the LiveKit connection URL and the token to connect to the LiveKit room.
```

Comment:
```suggestion
If successful, it will return the connection URL, which is composed of LiveKit's connection URL and the token to connect to the LiveKit room.
```

Reply:
Changed it!